scripts/sem_bleaching_map.py

```diff
@@ -463,101 +720,210 @@ def estimate_acq_time(self, shape):
-            try:
-                # TODO: support drift correction (cf ar_spectral_ph)
-                for i in numpy.ndindex(shape):
-                    logging.info("Acquiring pixel %d,%d", i[1], i[0])
-                    sem_data.append(self.bleach_spot(spots[i].tolist()))
-                    ccd_data = self.get_fluo_image()
+            ccd_data_to_save = ccd_data[ccd_roi_idx]
+
+            # Set up the drift correction (using the dwell time used for overview)
+            if anchor:
+                logging.info("Starting with pre-exposing the anchor region")
+                de = drift.AnchoredEstimator(self.ebeam, self.sed, anchor, self.dtsem)
+                for ii in range(13):
+                    de.acquire()
+
+                de = drift.AnchoredEstimator(self.ebeam, self.sed, anchor, self.dtsem)
+                px_iter = de.estimateCorrectionPeriod(self.dperiod, 1.0, self.gridpitch)
+                de.acquire()  # original anchor region
+                self.save_data(de.raw[-1], "%s%02d-driftAnchor-%05d%s" % (fn_base, 0, 0, fn_ext))
+                self.driftlog = numpy.array([0, 0])
+                self.drifttime = numpy.array(time.time())
+                self.driftlog.shape = (2, 1)
+
+                next_dc = px_iter.next()
+
+            for ii in range(cycles):
+                # TODO: fix this super-arbitrary dwell time change
+                if ii >= 1:
+                    self.ebeam.dwellTime.value = 2 * self.dt
+
+                n = 0
+                sem_data = []
+
+                # save the fluorescence without bleaching (or end of previous cycle)
+                self.save_data(ccd_data_to_save, "%s%02d-ccd-%05d%s" % (fn_base, ii, n, fn_ext))
+                n_drift = 0
+                for g in numpy.ndindex(self.gridpitch):
                     n += 1
-                    self.save_data(ccd_data, "%s-ccd-%05d%s" % (fn_base, n, fn_ext))
-                    ccd_count.append(self.ccd_image_to_count(ccd_data, ccd_roi))
-                
-                # TODO: acquire a "SEM survey" image?
-                 
+
+                    spotssubset = spots[g[0]::self.gridpitch[0], g[1]::self.gridpitch[1], :]
+                    subsetshape = spotssubset.shape[0:2]
+
+                    dur_frame = self.estimate_acq_time_gridscan(subsetshape)
+                    # overhead_est = 1 * dur_frame + 0.2
+                    overhead_est = 0
+                    # overhead_est = 1 * dur_frame + 0.5
+
+                    # self.ccd.exposureTime.value = dur_frame + overhead_est
+                    # logging.info("Setting exposuretime for spotsimage to: %s (includes %s overhead)",
+                    # units.readable_time((dur_frame + overhead_est)),
+                    # units.readable_time((overhead_est)))
+                    # self.ccd_acq_complete.clear()
+                    # self.ccd.data.subscribe(self.on_ccd_data)
+                    # time.sleep(0.2)
```

Comment:
Is this code supposed to be commented out? If it has functionality maybe write a comment above it with the purpose of it

Reply:
This is not written by me, so I didn't want to delete it. People that know how to use the script (like phd students in TU-Delft) use to comment and uncomment some parts of the script. Anyway, I will have to delete this pull-request and make a new one :)